node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -1731,53 +1826,75 @@ mod tests {
+        let balance_account_2 = multiply_by_quintillion(500);
+        let sketched_account_2 = SketchedPayableAccount {
+            wallet_addr_seed: "def",
+            balance_minor: balance_account_2,
+            threshold_intercept_major: multiply_by_billion(100),
+            permanent_debt_allowed_major: multiply_by_billion(30),
+        };
         // Thrown away as the second one due to a shortage in the service fee,
         // listed among accounts to disqualify and picked eventually for its
         // lowest weight
-        let balance_3 = multiple_by_billion(250_000_000_000);
-        let account_3 =
-            make_plucked_qualified_account("ghi", balance_3, 90_000_000_000, 20_000_000_000);
-        let qualified_payables = vec![account_1.clone(), account_2, account_3];
-        let analyzed_payables = convert_collection(qualified_payables);
+        let balance_account_3 = multiply_by_quintillion(250);
+        let sketched_account_3 = SketchedPayableAccount {
+            wallet_addr_seed: "ghi",
+            balance_minor: balance_account_3,
+            threshold_intercept_major: multiply_by_billion(90),
+            permanent_debt_allowed_major: multiply_by_billion(20),
+        };
+        let total_weight_account_1 = multiply_by_quintillion(900);
+        let total_weight_account_2 = multiply_by_quintillion(500);
+        let total_weight_account_3 = multiply_by_quintillion(750);
+        let sketched_accounts = [sketched_account_1, sketched_account_2, sketched_account_3];
+        let (analyzed_payables, actual_disqualification_limits) =
+            make_analyzed_accounts_and_show_their_actual_disqualification_limits(sketched_accounts);
         let calculator_mock = CriterionCalculatorMock::default()
-            .calculate_result(multiple_by_billion(900_000_000_000))
-            .calculate_result(multiple_by_billion(500_000_000_000))
-            .calculate_result(multiple_by_billion(750_000_000_000));
-        let mut subject = PaymentAdjusterReal::new();
-        subject.calculators = vec![Box::new(calculator_mock)];
-        subject.logger = Logger::new(test_name);
-        let service_fee_balance_in_minor = balance_1 - multiple_by_billion(10_000_000_000);
+            .calculate_result(total_weight_account_1)
+            .calculate_result(total_weight_account_2)
+            .calculate_result(total_weight_account_3);
+        let subject = PaymentAdjusterBuilder::default()
+            .start_with_inner_null()
+            .replace_calculators_with_mock(calculator_mock)
+            .logger(Logger::new(test_name))
+            .build();
+        let service_fee_balance_in_minor = balance_account_1 - multiply_by_quintillion(10);
         let agent_id_stamp = ArbitraryIdStamp::new();
         let agent = BlockchainAgentMock::default()
             .set_arbitrary_id_stamp(agent_id_stamp)
             .service_fee_balance_minor_result(service_fee_balance_in_minor);
+        let transaction_count_limit = 2;
```

Comment:
I believe this variable is quite important so please migrate it to the top lines. ✌️